### ISSUE TITLE
Publishing source files onto NPM 

### ISSUE THREAD

**Issue by Alan-Cha** [strongloop/oasgraph#38]
Currently, we are publishing the source files, specifically the Typescript files under `src/`, in our NPM module. Perhaps this is not the best strategy.

**Comment by ErikWittern** [comment]
Some immediate thoughts:

* We actually want to share typing information, because it allows developers who also use TypeScript to get checks when using OASGraph.
* We might use the `types` property of package.json to point to the source files, see for example [here](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html).

More inspiration can be found for example at: https://medium.com/cameron-nokes/the-30-second-guide-to-publishing-a-typescript-package-to-npm-89d93ff7bccd

**Comment by ErikWittern** [comment]
Here is the article explaining how to do it:

https://medium.com/@jdxcode/for-the-love-of-god-dont-use-npmignore-f93c08909d8d

TLDR: Use `files` attribute in `package.json` to whitelist files, rather than `.npmignore` (which will cause npm to _ignore_ .gitignore).

**Comment by ErikWittern** [comment]
As of June 10th, excluding the sources would make the package 150 KB smaller. But it would also prohibit our users from getting references to the code if something goes wrong (using source maps), and it will generally make it hard for users to study the source code.

I.e., for now, we decide against this.
